### PR TITLE
Add tests for global configuration and support mail tests

### PR DESCRIPTION
**Pull request by laoneo**
### Summary of Changes
Adds new tests for the global configuration. Additionally it adds some mailing tasks to test mails.

### CODE REVIEW

File: tests/System/plugins/db.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
+const mysql = require('mysql');
+const postgres = require('postgres');
+
+// Rows cache of items which got inserted
```
Comment (brianteeman) [inline tests/System/plugins/db.js]: this does not make sense. what do you mean?

File: tests/System/plugins/db.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+      if (insertItem && result.length && result[0].id) {
+        insertItem.rows.push(result[0].id);
+      }
+
+      // Normalize the object
+      return { insertId: result[0].id };
+    });
+  }
+
+  // Return a promise when resolves the query
```
Comment (brianteeman) [inline tests/System/plugins/db.js]: This does not make sense. what do you mean?

File: tests/System/plugins/mail.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+ *
+ * @returns null
+ */
+function startMailServer(config) {
+  // Check if the mail server is already started
+  if (mailServer !== null) {
+    return null;
+  }
+
+  // Start he mail server on the cofnigured port
```
Comment (brianteeman) [inline tests/System/plugins/mail.js]:
```suggestion
  // Start the mail server on the configured port
```

File: tests/System/plugins/db.js (hunk 1 above)
Comment (laoneo) [reply tests/System/plugins/db.js]: Its the cache of items which are added by an insert statement

File: tests/System/plugins/db.js (hunk 2 above)
Comment (brianteeman) [reply tests/System/plugins/db.js]: Do you mean
Return a promise when the query is resolved

Comment (laoneo) [reply tests/System/plugins/db.js]: More of that it returns a promise which runs the given query

**Pull request merged**